services/uibackend/main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+)
+
+type Onboarding struct {
+	Token string `json:"token"`
+}
+
+func parseKey(s []byte) *rsa.PrivateKey {
+
+	Block, _ := pem.Decode(s)
+	k, err := x509.ParsePKCS1PrivateKey(Block.Bytes)
+	if err != nil {
+		panic(err)
+	}
+	return k
+}
+
+func generateToken(w http.ResponseWriter, r *http.Request) {
+
+	id := uuid.New()
+	now := time.Now()
+	payload := fmt.Sprintf("%s%s%s%d%s", "{\"id\":\"", id.String(), "\", \"expirationDate\":\"", now.Unix()+172800, "\"}")
+	encodedPayload := base64.StdEncoding.EncodeToString([]byte(payload))
+
+	msg := []byte(payload)
+
+	// Before signing, we need to hash our message
+	// The hash is what we actually sign
+	msgHash := sha256.New()
+	_, err := msgHash.Write(msg)
+	if err != nil {
+		os.Exit(1)
+	}
+	msgHashSum := msgHash.Sum(nil)
+
+	pemString, err := os.ReadFile("/etc/onboarding-private-key/key")
+
+	// In order to generate the signature, we provide a random number generator,
+	// our private key, the hashing algorithm that we used, and the hash sum
+	// of our message
+
+	signature, err := rsa.SignPKCS1v15(rand.Reader, parseKey(pemString), crypto.SHA256, msgHashSum)
+	if err != nil {
+		os.Exit(1)
```

Comment:
Shouldn't we log an error here? also why are you exiting the entire server? shouldn't we just have an HTTP reply with an Interal error code?